chapters/chap02-sec05.tex

```diff
@@ -68,5 +69,27 @@ \section{Processing $\zethTx$}\label{zeth-protocol:process-tx}
                     \item $\mixer.\balance = \mixer.\balance - \vout$
                 \end{itemize}
             \item Emit an event (\cref{sssec:ethereum-events}) $\evMixOut$ of type \mixEventDType, containing the new root $\mkroot'$ of the Merkle tree of commitments, the nullifiers $\indexedset{\nf{i}}{i \in [\jsin]}$, commitments to the newly created \zethnotes $\inp{\mix}.\primInp.\cms{}$, and the corresponding ciphertexts $\inp{\mix}.\primInp.\ciphers$.
-        \end{itemize}
+        \end{enumerate}
 \end{enumerate}
+
+\begin{remark}\label{zeth-protocol:process-tx:dispatch-call}
+  In some deployments, verification of the zk-SNARK proof may be delegated to some trusted external mechanism, such that the proof $\zkp$ does not appear as public data on-chain. As an example, consider the case where multiple \zeth~transactions are aggregated by an external system, such as that described in \cite{rondelet2020zecale}. The multiple zk-SNARK proofs become auxiliary inputs to a ``wrapping'' zk-SNARK which checks their validity via a single proof verification. A modified version of the \mix~function receives \mix~parameters from a trusted contract without $\inp{\mix}.\zkp$.
```

Comment:
Same comment as above w.r.t "trusted external mechanism".

Reply:
(Same applies for "trusted contract". I would argue that the contract is not trusted at all. Anyone can verify - and should verify - the bytecode of the deployed zecale contract before binding it to their base app on-chain logic. As such, everyone can be sure that the right checks are carried out, and thus there is no need to " trust" the zecale contract and "hope it behaves well" here.)

Reply:
Stated otherwise, people may want to somehow trade checks in the protocol for more trust, but this is not  requirement (i.e. the delegation scheme can be trustless)

Reply:
Updated to avoid confusion. (As above, no implication that the mechanism and contract are not verified. Implicitly, users would only trust them after satisfying themselves that that have the correct behaviour.)